glide-core/src/client/mod.rs

```diff
@@ -59,6 +62,58 @@ pub const DEFAULT_MAX_INFLIGHT_REQUESTS: u32 = 1000;
 /// and performance overhead.
 pub const CONNECTION_CHECKS_INTERVAL: Duration = Duration::from_secs(3);
 
+/// A static Glide runtime instance
+static RUNTIME: OnceCell<GlideRt> = OnceCell::new();
+
+pub struct GlideRt {
+    pub runtime: Handle,
+    shutdown_notifier: Arc<Notify>,
+}
+
+/// Initializes a single-threaded Tokio runtime in a dedicated thread (if not already initialized)
+/// and returns a static reference to the `GlideRt` wrapper, which holds the runtime handle and a shutdown notifier.
+/// The runtime remains active indefinitely until a shutdown is triggered via the notifier, allowing tasks to be spawned
+/// throughout the lifetime of the application.
+pub fn get_or_init_runtime() -> Result<&'static GlideRt, String> {
+    RUNTIME.get_or_try_init(|| {
+        let notify = Arc::new(Notify::new());
+        let notify_thread = notify.clone();
+        let (tx, rx) = oneshot::channel();
+
+        thread::Builder::new()
+            .name("glide-runtime-thread".into())
+            .spawn(move || {
+                match Builder::new_current_thread().enable_all().build() {
+                    Ok(runtime) => {
+                        let handle = runtime.handle().clone();
```

Comment:
If GlideTt contains Runtime then:
```suggestion
                        let glide_rt = GlideRt {
                            runtime: runtime,
                            shutdown_notifier: notify_thread.clone(),
                        };
```

Reply:
the runtime needs to be used right after (runtime.block_on), so moving the runtime itself to the GlideRt struct requires additional API to get the runtime handle - which is basically a reference counter to the runtime. So why not storing the reference countered runtime to begin with?